Lib/test/test_functools.py

```diff
@@ -249,6 +249,12 @@ def test_placeholders_optimization(self):
         self.assertEqual(p2.args, (PH, 0))
         self.assertEqual(p2(1), ((1, 0), {}))
 
+    def test_placeholders_kw_restriction(self):
+        PH = self.module.Placeholder
+        exc_string = 'keyword Placeholders are not allowed'
+        with self.assertRaisesRegex(TypeError, exc_string):
```

Comment:
Generally it is wise to loosen the regex for exception messages; they tend to get tweaked every now and then.

```suggestion
        with self.assertRaisesRegex(TypeError, "Placeholder"):
```